docs/src/cookbooks/cookbook.md

```diff
@@ -462,3 +464,52 @@ ChiselStage.emitVerilog(new Salt)
 ```scala mdoc:verilog
 ChiselStage.emitVerilog(new Salt)
 ```
+
+## Directionality
+
+### How do I strip directions from a bidirectional Bundle (or other Data)?
+
+Given a bidirectional port like a `Decoupled`, you will get an error if you try to connect it directly
+to a register:
+
+```scala mdoc:silent
+import chisel3.util.Decoupled
+class BadRegConnect extends Module {
+  val io = IO(new Bundle {
+    val enq = Decoupled(UInt(8.W))
+  })
+  
+  val monitor = RegNext(io.enq)
+}
+```
+
+```scala mdoc:crash
+ChiselStage.emitVerilog(new BadRegConnect)
+```
+
+While there is no construct to "strip direction" in Chisel3, wrapping a type in `Output(...)` will
+coerce the direction to `Output` (even if the input is bidirectional).
+This will have the desired result when used to construct a Register:
+
+```scala mdoc:silent
+import chisel3.util.Decoupled
+class CoercedRegConnect extends Module {
+  val io = IO(new Bundle {
+    val enq = Flipped(Decoupled(UInt(8.W)))
+  })
+  
+  // Make a Reg of the correct type instead of using RegNext
```

Comment:
I think this example isn't very clear without the resulting verilog. Is this saying that you will only make registers for the outputs? Or will you get a register for the `monitor.valid` as well?

Reply:
I didn't include the Verilog because of all of the RANDOMIZATION stuff which is pretty distracting 🙁

Reply:
maybe just a comment then that makes it clear you *will* get a Reg for the `valid` as well?